app/src/newest/java/net/gini/android/gvlexample/gini/ExampleEventTracker.java

```diff
@@ -72,6 +73,6 @@ private String printEventDetails(@NonNull final Map<String, String> details) {
     private void trackEvent(@NonNull final Event<?> event) {
         final String name =
                 event.getType().getClass().getSimpleName() + "." + event.getType().name();
-        Analytics.trackEvent(name, event.getDetails());
+        Analytics.trackEvent(name, (EventProperties) event.getDetails());
```

Comment:
I think this will not work. The incoming events have `Map<String, Object>` as details property type. I think we need to convert the `Map<String, Object>` details map into a `Map<String, String> map.

Reply:
`    private void trackEvent(@NonNull final Event<?> event) {
        final String name =
                event.getType().getClass().getSimpleName() + "." + event.getType().name();
        final Map<String, String> properties = toProperties(event.getDetails());
        Analytics.trackEvent(name, properties);
    }

    private Map<String, String> toProperties(@NonNull final Map<String, Object> details) {
        final Map<String, String> properties = new HashMap<>(details.size());
        for (Entry<String, Object> entry: details.entrySet()) {
            final String value = entry.getValue().toString();
            properties.put(entry.getKey(), value);
        }
        return properties;
    }`

Reply:
Thank you for the snippet.